src/benchq/algorithms/lde_solver.py

```diff
@@ -265,3 +265,54 @@ def get_prep_int(
     prep_int_prime = transpile(prep_int_prime, basis_gates=["rz", "h", "cx"])
 
     return import_from_qiskit(prep_int), import_from_qiskit(prep_int_prime)
+
+
+def matrix_exponentiation(
+    be_matrix: Circuit, matrix_norm: float, time: float, beta: float, epsilon: float
+) -> Circuit:
+    """Constructs the quantum circuit for a single time step in the
+    time-marching algorithm - construct the approximation exp(A)=f(A).
+
+    The PREP_int unitary acts on log2(K) qubits starting from the second qubit.
+    The SEL_inv unitary acts on all qubits.
+    The PREP_int_prime_dag unitary acts on log2(K) qubits starting from
+    the second qubit.
+
+    The unitaries are appended in the following order
+    PREP_int * SEL_inv * PREP_int_prime_dag.
+
+    Args:
+        be_matrix (Circuit): the block encoding of a matrix.
+        matrix_norm (float): the norm of the matrix that is
+            to be block encoded.
+        time (float): the time interval one seeks solution
+            for a differntial equation.
+        beta (float): a bounding parameter of the matrix.
```

Comment:
Bounding parameter isn't very clear. I might want to say, "an upper bound for the largest eigenvalue of the block encoding"